src/viam/sdk/common/service_helper.cpp

```diff
@@ -5,6 +5,7 @@
 namespace viam {
 namespace sdk {
 
+// NOLINTBEGIN(bugprone-exception-escape)
```

Comment:
For some reason we were getting complaints that these could throw errors, which caused `clang-tidy` to complain because of the `noexcept` marking. This is false though. I wasn't sure how to convince `clang-tidy` it was wrong so I just disabled the `bugprone-exception-escape` linting for this file.

Reply:
Weird... I think disabling is fine; I see other [SO posts](https://stackoverflow.com/questions/61014184/clang-tidys-bugprone-exception-escape-behaves-weirdly-with-msvc-stl) about that linter being a little buggy.

Reply:
I feel like at a certain point you just want to let the compiler enforce `noexcept` by calling `terminate`. If you are at the point where you cant construct the status object safely, well, it's time to bail out.